src/main/java/de/tum/in/www1/artemis/web/rest/ExerciseHintResource.java

```diff
@@ -0,0 +1,175 @@
+        return ResponseEntity.created(new URI("/api/exercise-hints/" + result.getId()))
+                .headers(HeaderUtil.createEntityCreationAlert(applicationName, true, ENTITY_NAME, result.getId().toString())).body(result);
+    }
+
+    /**
+     * {@code PUT  /exercise-hints/{id}} : Updates an existing exerciseHint.
+     *
+     * @param exerciseHint the exerciseHint to update.
+     * @return the {@link ResponseEntity} with status {@code 200 (OK)} and with body the updated exerciseHint,
+     * or with status {@code 400 (Bad Request)} if the exerciseHint is not valid,
+     * or with status {@code 500 (Internal Server Error)} if the exerciseHint couldn't be updated.
+     * @throws URISyntaxException if the Location URI syntax is incorrect.
+     */
+    @PutMapping("/exercise-hints/{id}")
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<ExerciseHint> updateExerciseHint(@RequestBody ExerciseHint exerciseHint, @PathVariable Long id) {
+        log.debug("REST request to update ExerciseHint : {}", exerciseHint);
+        if (exerciseHint.getId() == null || !id.equals(exerciseHint.getId()) || exerciseHint.getExercise() == null) {
+            return badRequest();
+        }
+        Optional<ExerciseHint> hintBeforeSaving = exerciseHintService.findOne(id);
+        if (!hintBeforeSaving.isPresent()) {
+            return notFound();
+        }
+        if (!authCheckService.isAtLeastTeachingAssistantForExercise(exerciseHint.getExercise())
+                || !authCheckService.isAtLeastTeachingAssistantForExercise(hintBeforeSaving.get().getExercise())) {
+            return forbidden();
+        }
+        ExerciseHint result = exerciseHintService.save(exerciseHint);
+        return ResponseEntity.ok().headers(HeaderUtil.createEntityUpdateAlert(applicationName, true, ENTITY_NAME, exerciseHint.getId().toString())).body(result);
+    }
+
+    /**
+     * {@code GET  /exercise-hints/:id} : get the "id" exerciseHint.
+     *
+     * @param id the id of the exerciseHint to retrieve.
+     * @return the {@link ResponseEntity} with status {@code 200 (OK)} and with body the exerciseHint, or with status {@code 404 (Not Found)}.
+     */
+    @GetMapping("/exercise-hints/{id}")
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<ExerciseHint> getExerciseHint(@PathVariable Long id) {
+        log.debug("REST request to get ExerciseHint : {}", id);
+        Optional<ExerciseHint> exerciseHint = exerciseHintService.findOne(id);
+        if (!exerciseHint.isPresent()) {
+            return notFound();
+        }
+        if (!authCheckService.isAtLeastTeachingAssistantForExercise(exerciseHint.get().getExercise())) {
+            return forbidden();
+        }
+        return ResponseEntity.ok().body(exerciseHint.get());
+    }
+
+    /**
+     * {@code GET  /exercise-hints/by-exercise-id/:id} : get the exerciseHints of a provided exercise.
```

Comment:
please change this part as it is different from GetMapping

Reply:
You are right, I changed it!